### PR TITLE
Add baseline & wandb log

### PR DESCRIPTION
**Pull request by dddkhoa**

### CODE REVIEW

File: Scaled_Attention/main_train.py [hunk 1]
```diff
@@ -38,6 +38,7 @@ def get_args_parser():
     parser.add_argument('--epochs', default=300, type=int)
     parser.add_argument('--pi_reg', type= bool, default= False)
     parser.add_argument('--pi_reg_coef', type=float, default= 0.1)
+    parser.add_argument('--is_baseline', action= "store_true", default=False)
```
Comment (thanhquangtran) [inline Scaled_Attention/main_train.py]: should be `attention_type` or something like that.

File: Scaled_Attention/main_train.py [hunk 2]
```diff
@@ -394,17 +396,17 @@ def main(args):
...
+        project="implicit-attention",
         
         # track hyperparameters and run metadata
         config=args)
-        wandb.run.name = "run_name"
+        wandb.run.name = f"{args.model}-{args.batch_size}-{args.lr}--{args.data_set}-{args.is_baseline}"
 
-    if args.eval:
-        test_stats = evaluate(data_loader_val, model, device, wandb=args.wandb)
+    if args.eval and args.gpu == 0:
```
Comment (thanhquangtran) [inline Scaled_Attention/main_train.py]: where can I find args.gpu?

File: Scaled_Attention/main_train.py [hunk 3]
```diff
@@ -453,6 +456,9 @@ def main(args):
                      'epoch': epoch,
                      'n_parameters': n_parameters,
                      'max_accuracy': max_accuracy}
+        
+        if args.wandb and global_rank == 0:
```
Comment (thanhquangtran) [inline Scaled_Attention/main_train.py:460]: consider using MetricLogger

File: Scaled_Attention/engine.py [hunk 4]
```diff
@@ -82,14 +82,14 @@ def train_one_epoch(model: torch.nn.Module, criterion: DistillationLoss,
     # gather the stats from all processes
     metric_logger.synchronize_between_processes()
     print("Averaged stats:", metric_logger)
-    if wandb:
+    if use_wandb and rank==0:
```
Comment (thanhquangtran) [inline Scaled_Attention/engine.py:85]: please use metric logger

File: Scaled_Attention/main_train.py (hunk 3 above)
Comment (thanhquangtran) [reply Scaled_Attention/main_train.py:460]: also, consider only logging stats here and remove wandb in `def evaluate`

File: Scaled_Attention/main_train.py (hunk 1 above)
Comment (dddkhoa) [reply Scaled_Attention/main_train.py]: updated

File: Scaled_Attention/main_train.py (hunk 2 above)
Comment (dddkhoa) [reply Scaled_Attention/main_train.py]: in `Scaled_Attention/utils.py`, `args.gpu = int(os.environ['LOCAL_RANK'])`, nevermind, I'm changing to use `global_rank (=utils.get_rank())` instead

File: Scaled_Attention/engine.py (hunk 4 above)
Comment (dddkhoa) [reply Scaled_Attention/engine.py:85]: can u elaborate? right below this line I'm using wandb to log metrics from metric logger

File: Scaled_Attention/main_train.py (hunk 3 above)
Comment (dddkhoa) [reply Scaled_Attention/main_train.py:460]: already removed wandb in evaluate

Comment (dddkhoa) [reply Scaled_Attention/main_train.py:460]: MetricLogger returns the stats for metrics, then I use wandb to log those, or do u mean call wandb inside MetricLogger?

File: Scaled_Attention/engine.py (hunk 4 above)
Comment (thanhquangtran) [reply Scaled_Attention/engine.py:85]: okay I think I misunderstood metricLogger

**Review by thanhquangtran — APPROVED** [review]
